### DISCUSSION TITLE
Feature Request(s)/other suggested improvements

### DISCUSSION THREAD

**Discussion by mjxbenet** [edemaine/cocreate#146] [Ideas]
I love cocreate and use it for most of my private tutoring/classes. If I have time later, I might list out all the features I like, but for now, here's a few I think would be good additions:

1. Customizable keyboard shortcuts (e.g. change pencil shortcut from 'p' to 'w')
2. Keyboard shortcuts for colors
3. Ability to delete pages
4. Ability to switch from infinite grid to finite space (Google Jam boards does this, I believe). Alternatively, a way to see the viewing area of other people on the board
5. Ability to customize the cocreate url
6. Ability to restrict access to a url
7. Ability to flatten multiple items into one 
     a) If i write a bunch of text, select it all, and move it. It takes a long time for others to see it
8. When selecting things and moving them, you have to click very precisely on the exact line, or it unselects everything that's selected. I'm not sure how to fix this, perhaps a key that could be held to tell cocreate you want to start dragging the selection?

**Comment by edemaine** [comment]
Thanks for your thoughtful list! Many of these are already on the issues list, but some are not. Here's a point-by-point reply:

1. #57. The hardest part here is coming up with a settings interface (unless you have an idea for how to build keyboard shortcut customization into the main interface...?). That might wait for a possible port to React, which would make interface building easier.
2. Hmm, I hadn't thought of that, but it sounds like a useful idea. Do you have an idea for what those shortcuts would look like? Do you know any other programs that do this? Or are you just imagining a few favorite colors assigned single characters by the user?
3. #18. This is a high priority (very frequently requested), and will be implemented as part of #137, which is likely the next big thing I'll take on. To be clear, it won't be possible to delete pages, but it will be possible to archive (hide) them.
4. #19. Page boundaries are something I've always wanted, as I like writing lecture notes that can be easily printed. It's probably also not hard... just hasn't been a priority yet. For seeing what others can see, #5 has the idea of tracking "view rectangles"; the only issue is when to show them. Probably it could be part of #137.
5. This has been discussed (also for Comingle), but I think it's extremely risky (at least until we have accounts/authentication) because it makes for a huge security hole. Knowing the room name/URL means you can make a mess of the room, so if you choose an easy-to-guess name, bad things are pretty likely.  Then again, Jitsi and whenisgood let you choose your own names, so maybe it's not crazy... but the impact a bad user can have on a Cocreate board is perhaps more severe.
6. #4. The main thing holding me back here has been not wanting to have yet another user authentication system (as I mostly use Cocreate in the context of Coauthor and Comingle, and Coauthor has its own authentication). But especially for those using Cocreate directly, it probably makes sense... #85 is another relevant request, if that helps for you purpose.
7. Do you mean grouping objects? That's interesting, but technically very difficult to do, and I'm not really sure it works well in the context of multiple people editing the same objects. But the performance of moving multiple objects is indeed a concern. I am already taking care to make that efficient, but I agree that it doesn't feel great, though I'm not sure what else I can do.  It's maybe worth noting that I recently added the ability to write many lines of text in one text object, so that is one way to speed up your use case.
8. #109. My plan is to just add a bigger tolerance so you don't need to be right on the objects to drag them. This is annoying to fix for technical reasons (#87).

If you're a developer, you're certainly welcome to work on any of these!

**Reply by mjxbenet** [reply]
Thanks for the quick and detailed reply, and apologies for the delay in my response. I have limited coding experience, so forgive me if I make an ignorant assumption.

1. and 2. https://cjquines.com/qboard/ does both of these things, though it's "meant for giving lectures
or scratch work, and as such it explicitly doesn't support collaboration or lots of colors, or more complicated tools like text" per the creator (CJ Quines, MIT sophomore).
1. just a personal note: I know not everyone is like me, but I prefer keeping keyboard shortcuts on the left half of the keyboard, as I could keep one hand on the keyboard and write with the other 
2.  I don't know of any programs that do this. I'm imagining a few favorite colors assigned single characters by the user. This is a big QOL change for me personally - my teaching quality and efficiency is higher with cocreate, but switching colors (while still faster than Zoom) is the biggest bottleneck. 
5. One potential solution is to add a 'use at your own risk' disclaimer.
 A possible concern that could arise is people using this as a website?? a purposefully memorable site that they publicize/put something undesirable.
6.  I very much like the idea of the account with saved login (#4) 
7. and 8. I like the way Zoom does this: when items are selected, it gives the option to resize the selected items, and moving them around is pretty easy. Is something like that possible?

**Comment by ucalyptus2** [comment]
hi @edemaine, how do I copy image from cocreate into google slides?
And how to do vice-versa?

**Reply by edemaine** [reply]
The short answer is "you can't".  But some things are possible:

You can download SVG from Cocreate, using the [Download SVG](https://github.com/edemaine/cocreate/tree/master/doc#-download-svg) button in the bottom left.  PowerPoint supports SVG, but Google Slides does not seem to, directly.  See [this page](https://phdesign.com.au/general/adding-font-awesome-icons-to-google-slides/).

To go in the other direction, you'd have to get your slides into SVG format (not quite sure the best way to do that -- maybe via PDF), upload them to a website, and then you could embed them into Cocreate.  Won't be able to edit, but will be able to draw on top.
